src/tests/fs_tests.rs

```diff
@@ -805,44 +805,255 @@ pub mod fs_tests {
     }
 
     #[test]
```

Comment:
exhaustive tests overall, great job! Just wondering about an extra test case that may be interesting involving linking paths and then forking the process and unlinking one of the paths? What do you think?